src/sentry/incidents/logic.py

```diff
@@ -1318,6 +1325,28 @@ def get_alert_rule_trigger_action_msteams_channel_id(
     return channel_id
 
 
+def get_alert_rule_trigger_action_discord_channel_id(
+    channel_id,
+    organization,
+    integration_id,
+    **kwargs,
+) -> str:
+    from sentry.integrations.discord.utils.channel import validate_channel_id
+
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if integration is None:
+        raise InvalidTriggerActionError("Discord server is a required field.")
```

Comment:
Why does an integration not being found raise an error about a Discord server field?

Reply:
Yeah, that doesn't make much sense. I think Integration.DoesNotExist is more appropriate here.